src/algorithms.rs

```diff
@@ -80,6 +81,86 @@ impl Algorithm {
             Algorithm::EdDSA => AlgorithmFamily::Ed,
         }
     }
+
+    /// Converting Key Algorithm to Algorithm
+    pub fn frm_key_alogorithm(s: &KeyAlgorithm) -> Result<Self> {
```

Comment:
Thanks for the pr, I couldn't figure out how to best implement this myself, but - I think this should spell "from_key_algorithm"? And maybe implementing From<KeyAlgorithm> would be nice too, but I'm not sure

Reply:
Thanks for the comment. From trait signature would not match with us, I think From makes sense when we expect implicit conversion which will work almost always. Here the conversion is explicit and result in an error.

Btw renaming `frm_key_alogorithm` to `from_key_alogorithm`.